lib/block-supports/typography.php

```diff
@@ -451,51 +458,85 @@ function gutenberg_get_typography_font_size_value( $preset, $should_use_fluid_ty
+	$preferred_size = gutenberg_get_typography_value_and_unit( $font_size['size'] );
 
 	// Protect against unsupported units.
 	if ( empty( $preferred_size['unit'] ) ) {
-		return $preset['size'];
+		return $font_size['size'];
 	}
 
-	// If no fluid min or max font sizes are available, create some using min/max font size factors.
+	// If no fluid min font size is available, create one using min font size factor.
 	if ( ! $minimum_font_size_raw ) {
-		$minimum_font_size_raw = ( $preferred_size['value'] * $default_minimum_font_size_factor ) . $preferred_size['unit'];
+		$minimum_font_size_raw = round( $preferred_size['value'] * $default_minimum_font_size_factor, 3 ) . $preferred_size['unit'];
 	}
 
+	// Parses the minimum font size limit, so we can perform checks using it.
+	$minimum_font_size_limit = gutenberg_get_typography_value_and_unit(
+		$default_minimum_font_size_limit,
+		array(
+			'coerce_to' => $preferred_size['unit'],
+		)
+	);
+
+	/*
+	 * Checks if a user-defined font size is lower than $minimum_font_size_limit. If so than it should become the minimum font size.
+	 * Otherwise, if the minimum font size is lower than $minimum_font_size_limit
+	 * use $minimum_font_size_limit instead.
+	 */
+	if ( ! empty( $minimum_font_size_limit ) ) {
+		// @TODO check if there was no min size value passed!
+		if ( $preferred_size['value'] < $minimum_font_size_limit['value'] ) {
+			$minimum_font_size_raw = implode( '', $preferred_size );
+		} else {
+			$minimum_font_size_parsed = gutenberg_get_typography_value_and_unit(
+				$minimum_font_size_raw,
+				array(
+					'coerce_to' => $preferred_size['unit'],
+				)
+			);
+
+			if ( ! empty( $minimum_font_size_parsed ) && $minimum_font_size_parsed['value'] < $minimum_font_size_limit['value'] ) {
+				$minimum_font_size_raw = implode( '', $minimum_font_size_limit );
+			}
+		}
+	}
+
+	// If no fluid max font size is available, create one using max font size factor.
 	if ( ! $maximum_font_size_raw ) {
-		$maximum_font_size_raw = ( $preferred_size['value'] * $default_maximum_font_size_factor ) . $preferred_size['unit'];
+		$maximum_font_size_raw = round( $preferred_size['value'] * $default_maximum_font_size_factor, 3 ) . $preferred_size['unit'];
 	}
 
 	$fluid_font_size_value = gutenberg_get_computed_fluid_typography_value(
 		array(
-			'minimum_viewport_width' => $default_minimum_viewport_width,
-			'maximum_viewport_width' => $default_maximum_viewport_width,
-			'minimum_font_size'      => $minimum_font_size_raw,
-			'maximum_font_size'      => $maximum_font_size_raw,
-			'scale_factor'           => $default_scale_factor,
+			'minimum_viewport_width'  => $default_minimum_viewport_width,
+			'maximum_viewport_width'  => $default_maximum_viewport_width,
+			'minimum_font_size'       => $minimum_font_size_raw,
+			'maximum_font_size'       => $maximum_font_size_raw,
+			'scale_factor'            => $default_scale_factor,
+			'minimum_font_size_limit' => $default_minimum_font_size_limit,
```

Comment:
I looks like the minimum limit has already been applied, does it still need to be passed to this array?

Reply:
It does not. 

Let's keep this between us 😆